### PR TITLE
Update the Python SDK

### PR DESCRIPTION
**Pull request by davidiw**
* Make all tests pass
* Makes main work again
* Adds support for new token standard

---
This change is [Reviewable](https://reviewable.io/reviews/aptos-labs/aptos-core/2391)

### CODE REVIEW

File: ecosystem/python/sdk/README.md [hunk 1]
```diff
@@ -42,7 +52,7 @@ Currently the team is moving forward with pure python, but leaves the following
 ```bash
 npm install  -g @apidevtools/swagger-cli
 swagger-cli bundle --dereference ../../../api/doc/v0/openapi.yaml  -t yaml > openapi.yaml
-python -m openapi_python_client generate --path openapi.yaml
+python3 -m openapi_python_client generate --path openapi.yaml
```
Comment (gregnazario) [inline ecosystem/python/sdk/README.md:55]: Can't comment below.  Are you supposed to be committing pycache?

File: ecosystem/python/sdk/README.md [hunk 2]
```diff
@@ -18,16 +18,26 @@ Currently this is still in development and is unsuitable for directly interfacin
...
 
+To run all unit tests:
 ```bash
 python3 -m unittest discover -s aptos_sdk/ -p '*.py'
 ```
 
+To run all end-to-end tests:
+```bash
+ln -s `pwd`/aptos_sdk examples/
+python3 aptos_sdk/coin.py
```
Comment (gregnazario) [inline ecosystem/python/sdk/README.md:29]: Okay so we're only supporting python3 and being explicit?

File: ecosystem/python/sdk/aptos_sdk/transactions.py [hunk 3]
```diff
@@ -592,9 +592,11 @@ def verify_transactions(
         ser.struct(raw_transaction_generated)
         raw_transaction_generated_bytes = ser.output().hex()
 
+        print(raw_transaction_generated_bytes)
         ser = Serializer()
         ser.struct(signed_transaction_generated)
         signed_transaction_generated_bytes = ser.output().hex()
+        print(signed_transaction_generated_bytes)
```
Comment (gregnazario) [inline ecosystem/python/sdk/aptos_sdk/transactions.py:599]: Do you still want these prints?

File: ecosystem/python/sdk/examples/coin.py [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+    faucet_client = FaucetClient(FAUCET_URL, rest_client)
+
+    alice = Account.generate()
+    bob = Account.generate()
+
+    print("\n=== Addresses ===")
+    print(f"Alice: {alice.address()}")
+    print(f"Bob: {bob.address()}")
+
+    faucet_client.fund_account(alice.address(), 1_000_000)
```
Comment (gregnazario) [inline ecosystem/python/sdk/examples/coin.py:19]: Can we change this down to 5k?  I believe we changed the other examples

File: ecosystem/python/sdk/examples/token.py [hunk 5]
```diff
@@ -0,0 +1,99 @@
...
+    bob = Account.generate()
+
+    collection_name = "Alice's"
+    token_name = "Alice's first token"
+
+    print("\n=== Addresses ===")
+    print(f"Alice: {alice.address()}")
+    print(f"Bob: {bob.address()}")
+
+    faucet_client.fund_account(alice.address(), 10_000_000)
```
Comment (gregnazario) [inline ecosystem/python/sdk/examples/token.py:22]: Similarly, I think we should keep it under the default fund amount

File: ecosystem/python/sdk/aptos_sdk/bcs.py [hunk 6]
```diff
@@ -155,6 +155,11 @@ def map(
             self.fixed_bytes(key)
             self.fixed_bytes(value)
 
+    def sequence_serializer(
+        value_encoder: typing.Callable[[Serializer, typing.Any], bytes],
+    ):
+        return lambda self, values: self.sequence(values, value_encoder)
+
     def sequence(
         self,
```
Comment (gregnazario) [inline ecosystem/python/sdk/aptos_sdk/bcs.py:164]: Interesting, is this so you can delay serialization?

**Review by gregnazario — APPROVED** [review]
Please align on cached files and the tutorial / example numbers.

File: ecosystem/python/sdk/examples/coin.py (hunk 4 above)
Comment (davidiw) [reply ecosystem/python/sdk/examples/coin.py:19]: Right, need to add that.

File: ecosystem/python/sdk/aptos_sdk/bcs.py (hunk 6 above)
Comment (davidiw) [reply ecosystem/python/sdk/aptos_sdk/bcs.py:164]: Yeah, we use this for specifying how to serialize a transaction argument in a fancy way... I am a fancy guy after all

File: ecosystem/python/sdk/README.md (hunk 1 above)
Comment (davidiw) [reply ecosystem/python/sdk/README.md:55]: Hmm... I'll revisit the stack... they must have snuck in and then I removed them at the end :)

File: ecosystem/python/sdk/README.md (hunk 2 above)
Comment (davidiw) [reply ecosystem/python/sdk/README.md:29]: well it was always only python3 and probably a pretty recent version... we'll leave this for figuring out later..

**Pull request merged**